packages/next/src/withTRPC.tsx

```diff
@@ -94,31 +95,32 @@ export function withTRPC<
         trpc?: TRPCPrepassProps;
       },
     ) => {
-      const [{ queryClient, trpcClient, ssrState, ssrContext }] = useState(
-        () => {
-          if (props.trpc) {
-            return props.trpc;
-          }
+      const [prepassProps] = useState(() => {
+        if (props.trpc) {
+          return props.trpc;
+        }
 
-          const config = getClientConfig({});
-          const queryClient = new QueryClient(config.queryClientConfig);
-          const trpcClient = trpc.createClient(config);
-          return {
-            queryClient,
-            trpcClient,
-            ssrState: opts.ssr ? ('mounting' as const) : (false as const),
-            ssrContext: null,
-          };
-        },
-      );
+        const config = getClientConfig({});
+        const queryClient = new QueryClient(config.queryClientConfig);
+        const trpcClient = trpc.createClient(config);
+        return {
+          abortOnUnmount: config.abortOnUnmount,
+          queryClient,
+          trpcClient,
+          ssrState: opts.ssr ? ('mounting' as const) : (false as const),
+          ssrContext: null,
+        };
+      });
 
+      const { queryClient, trpcClient, ssrState, ssrContext } = prepassProps;
       const hydratedState = trpc.useDehydratedState(
         trpcClient,
         props.pageProps?.trpcState,
       );
 
       return (
         <trpc.Provider
+          abortOnUnmount={(prepassProps as any).abortOnUnmount ?? false}
```

Comment:
if we want Next and React to always behave the same, we should probably not handle the default value here, but let the nullish value fall through to createReactQueryHooks and default it there

Reply:
great point I only was testing this with next. will implement all the suggestions